libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -353,7 +352,7 @@ abstract class DefaultApolloExtension(
     return project.tasks.register(ModelNames.checkApolloVersions()) {
       val outputFile = BuildDirLayout.versionCheck(project)
 
-      it.inputs.property("allVersions", Callable {
+      it.inputs.property("allVersions", project.provider {
```

Comment:
Interesting... Was the issue that the callable is called eagerly?

Reply:
Exactly! (Credits to @hsz 🙏). However this works locally but just noticed we still got the `ConcurrentModificationException` on the CI 😭

Reply:
IIRC `intellij-plugin` uses the published version of the plugin so we won't be able to tell before the next release if the fix actually worked.

Reply:
We have snapshots. 🤫

Reply:
> `intellij-plugin` uses the published version of the Gradle plugin

You're right! And my mistake - thought it was working locally but it's only because I ran with `--no-configuration-cache` :(

Reply:
I'd say the concurrent modification happens on the `configurations` collection because we're iterating all of them but `addAllLater` ([here](https://github.com/JetBrains/intellij-platform-gradle-plugin/blob/d95a86476b108a2899143586b987282e5dec0cc3/src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt#L208)) may create new configurations while iterating ([here](https://github.com/JetBrains/intellij-platform-gradle-plugin/blob/d95a86476b108a2899143586b987282e5dec0cc3/src/main/kotlin/org/jetbrains/intellij/platform/gradle/extensions/IntelliJPlatformDependenciesHelper.kt#L216))

Proposed fix to iterate a read-only copy of the configurations:

```diff
     private fun getDeps(configurations: ConfigurationContainer): List<String> {
-      return configurations.flatMap { configuration ->
+      // see https://github.com/apollographql/apollo-kotlin/pull/5657
+      val currentConfigurations = configurations.toList()
+      return currentConfigurations.flatMap { configuration ->
         configuration.dependencies
```

Means we won't check the newly created configurations for conflicting Apollo dependencies but it's probably fine.

Reply:
Of course this also requires a new release so just disabling the task inside `intellij-plugin` for now is the quickest solution.

Reply:
@hsz sorry, messages crossed. Regarding SNAPSHOTs, it's about Apollo releases. AKIJP (Apollo Kotlin IntelliJ Plugin) uses AKGP (Apollo Kotlin Gradle Plugin) . We could verify using Apollo snapshots but means we'd need a separate PR to check the plugin-ception 🙃 . 

All in all, my vote goes to disabling the check 😄

Reply:
Yeah, I figured too late. Anyway, I'm around if you need me!

Reply:
@martinbonnin unfortunately disabling the task doesn't work - the exception is thrown at configuration time - not 100% sure why, but looks like `getDeps` ends up being called during `org.gradle.configurationcache.ConfigurationCacheIO.writeConfigurationCacheState`.

Reply:
Mmmpff... Disable CC in that CI job?

Reply:
Haha that should work! And I'll open an PR for your suggested fix above, so it can be in the next release, hopefully fixing the issue.